cmds/live.c

```diff
@@ -415,30 +423,26 @@ static int forward_options(struct uftrace_opts *opts)
 
 int command_live(int argc, char *argv[], struct uftrace_opts *opts)
 {
-#define LIVE_NAME "uftrace-live-XXXXXX"
-	char template[32] = "/tmp/" LIVE_NAME;
 	int fd;
 	struct sigaction sa = {
 		.sa_flags = SA_RESETHAND,
 	};
 	int ret;
 
 	if (!opts->record) {
-		tmp_dirname = template;
+		snprintf(tmp_dirname, sizeof(tmp_dirname), "%s", TMP_LIVE_NAME);
 		umask(022);
-		fd = mkstemp(template);
+		fd = mkstemp(tmp_dirname);
 		if (fd < 0) {
-			/* can't reuse first template because it was trashed by mkstemp */
-			strcpy(template, LIVE_NAME);
-
 			if (errno != EPERM && errno != ENOENT)
-				pr_err("cannot access to /tmp");
+				pr_err("cannot access to " TMP_LIVE_NAME);
 
-			fd = mkstemp(template);
+			/* can't reuse first template because it was trashed by mkstemp */
+			snprintf(tmp_dirname, sizeof(tmp_dirname), "%s", LIVE_NAME);
+			fd = mkstemp(tmp_dirname);
 
 			if (fd < 0)
-				pr_err("cannot create temp name");
-			tmp_dirname = template;
+				pr_err("cannot create " LIVE_NAME);
```

Comment:
Ditto.

Reply:
Thank you @namhyung, I understand the `mkstemp would replace "XXXXXX"`. Done!